### PR TITLE
[SPARK-28536][SQL][PYTHON][TESTS] Reduce shuffle partitions in Python UDF tests in SQLQueryTestSuite

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

In Python UDF tests, the number of shuffle partitions matters considerably in the testing time because it requires to fork and communicate between external processes.

**Before:**

image

**After: (with 4)**

Screen Shot 2019-07-27 at 10 43 34 AM

## How was this patch tested?

Manually tested in my local.

**Before:**

```
[info] SQLQueryTestSuite:
[info] - udf/udf-window.sql - Scala UDF (58 seconds, 558 milliseconds)
[info] - udf/udf-window.sql - Regular Python UDF (58 seconds, 371 milliseconds)
[info] - udf/udf-window.sql - Scalar Pandas UDF (1 minute, 8 seconds)
```

**After:**

```
[info] SQLQueryTestSuite:
[info] - udf/udf-window.sql - Scala UDF (14 seconds, 690 milliseconds)
[info] - udf/udf-window.sql - Regular Python UDF (10 seconds, 467 milliseconds)
[info] - udf/udf-window.sql - Scalar Pandas UDF (10 seconds, 895 milliseconds)
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala [hunk 1]
```diff
@@ -279,6 +279,10 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
 
     testCase match {
       case udfTestCase: UDFTest =>
+        // In Python UDF tests, the number of shuffle partitions matters considerably in
+        // the testing time because it requires to fork and communicate between external
+        // processes.
+        localSparkSession.conf.set(SQLConf.SHUFFLE_PARTITIONS.key, 8)
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: Shall we use more smaller value?

Comment (HyukjinKwon) [reply sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: We could ..... I don't know lower than 8 is good enough though.. let's see if it really decreases the time first in Jenkins .. :-)

**Comment by HyukjinKwon** [comment]
Screen Shot 2019-07-27 at 3 48 46 PM

Now it looks much better and faster

**Comment by gatorsmile** [comment]
retest this please

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for updating.
With 4, every UTF tests goes under 37 secs. (Previously, it was 54 secs with 8).
I update the screenshot of the *AFTER* with the result of `4`.

Merged to master.

Thank you, @HyukjinKwon and @gatorsmile .

**Comment by HyukjinKwon** [comment]
Thank you @dongjoon-hyun

**Pull request closed without merging**